lib/http2/hpack.c

```diff
@@ -410,14 +485,18 @@ int h2o_hpack_parse_headers(h2o_req_t *req, h2o_hpack_header_table_t *header_tab
                             const char **err_desc)
 {
     const uint8_t *src_end = src + len;
+    int parsing_ret = 0;
```

Comment:
I think you could use test the value of `*err_desc` to determine the code that should be returned by this function.